django/db/backends/base/features.py

```diff
@@ -294,6 +294,9 @@ class BaseDatabaseFeatures:
     # Does the backend support boolean expressions in the SELECT clause?
     supports_boolean_expr_in_select_clause = True
 
+    # Does the backend support indexes on expressions?
+    supports_expression_indexes = True
```

Comment:
This feature flag should be documented under `Backwards incompatible changes` in `3.1.txt`. You can add something similar to what's in `2.2.txt` for partial indexes in the same section.

Reply:
How about making these two flags `False` by default, thus making the feature not causing backwards incompatible changes.